src/commandHandlers/roles.ts

```diff
@@ -1,35 +1,83 @@
 import { MessageEmbed } from 'discord.js';
 
 import { getRoles } from './guild.service';
-import config from '../config';
+import config, { selfAssignableRoles } from '../config';
 
 /**
  * This command lists the available roles on the discord server
  */
 export const command = async (arg: [string, string], embed: MessageEmbed) => {
   const roles = await getRoles();
-  const rolesList = roles
-    .filter((r) => !r.name.includes('everyone')) // Filter the default role everyone has access to
-    .map((discordRole) => {
-      const role = Object.values(config.ROLE).find(
-        (r) => r.name === discordRole.name
-      );
-      const roleDescription = role ? `- ${role.description}` : '';
-
-      return `\n• ${discordRole.toString()} ${roleDescription}`;
-    });
-
-  embed.setTitle('Available Roles')
-    .setDescription(`Here is the list of roles on this server. You can assign almost any role to yourself. Some of the roles are admin only or given to you via a condition!
-        ${rolesList}
-        Example of usage:
-        ^iam add javascript`);
+  const specificRole = arg[1];
 
+  if (specificRole) {
+    if (arg[1] === '-a' || arg[1] === 'all') {
+      const roleInfo = getRoleLists(roles, (r) => r.name.includes('everyone'));
+      embed.setTitle('Available Roles')
+        .setDescription(`Here is the list of all the roles on this server. You can assign almost any role to yourself. Some of the roles are admin only or given to you via a condition!\n
+        ${roleInfo[0].toString()}
+        ${roleInfo[1].toString()}\n
+        Assigning this role to yourself:
+        \`^iam add javascript\``);
```

Comment:
we could use destructuring here and use the names we had before 😃 
```suggestion
      const [rolesList, describedRoles] = getRoleLists(roles, (r) => r.name.includes('everyone'));
      embed.setTitle('Available Roles')
        .setDescription(`Here is the list of all the roles on this server. You can assign almost any role to yourself. Some of the roles are admin only or given to you via a condition!\n
        ${rolesList.toString()}
        ${describedRoles.toString()}\n
        Assigning this role to yourself:
        \`^iam add javascript\``);
```